### PR TITLE
toilet update: ghetto filtering

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
add: you can use the toilet as ghetto chemical filter
add: you can flush the toilet
add: toilet water and eau de toilette reagents
/:cl:
how to : put paper in toilet, add a beaker and then put the liquids in the toilet and alt click to separate 1 reagent into the beaker
ghetto filtering because there is no way to filter stuff as ghetto chemist

### CODE REVIEW

File: code/game/objects/structures/watercloset.dm [hunk 1]
```diff
@@ -109,19 +143,30 @@
 				return
 			w_items += I.w_class
 			to_chat(user, "<span class='notice'>You carefully place [I] into the cistern.</span>")
-
+			return
+	else if(istype(I, /obj/item/paper))
+		centrifuge = TRUE
+		to_chat(user, "<span class='notice'>You stick some fitering papers inside</span>")
```
Comment (tralezab) [inline code/game/objects/structures/watercloset.dm]: fitering papers 😄

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -1929,3 +1929,29 @@
 	color = "#E6E6DA"
 	taste_mult = 0
 
+/datum/reagent/water/toilet
+	name = "toilet water"
+	description = "99% water."
+	color = "4f919f" // 79, 145, 159
+	taste_description = "toilet water"
+	glass_name = "glass of water"
+	glass_desc = "It seems normal water..."
```
Comment (tralezab) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: oog think "it seems normal water"

**Comment by HomocidlePancakes** [comment]
How the fuck do you filter shit with a sheet of paper and a toilet?

File: code/game/objects/structures/watercloset.dm [hunk 3]
```diff
@@ -6,21 +6,51 @@
...
+	. = ..()
+	if(centrifuge)
+		. += "<span class='notice'>Insert a beaker, fill the chemical compost you want to filter and flush to transfer one type of reagent into the beaker.</span>"
 
+/obj/structure/toilet/AltClick(mob/living/user)
+	if(!reagents.total_volume)
+		to_chat(user, "<span class='warning'>There is nothing inside!</span>")
+		return
+	if(fwoosh_cooldown < world.time)
+		//add sound here
```
Comment (tralezab) [inline code/game/objects/structures/watercloset.dm]: this

**Review by tralezab — COMMENTED** [review]
haha he said the word wrong!

File: code/game/objects/structures/watercloset.dm [hunk 4]
```diff
@@ -109,19 +143,30 @@
...
-		RG.reagents.add_reagent(/datum/reagent/water, min(RG.volume - RG.reagents.total_volume, RG.amount_per_transfer_from_this))
-		to_chat(user, "<span class='notice'>You fill [RG] from [src]. Gross.</span>")
-	else
-		return ..()
+		if(!output && centrifuge)
+			if(user.transferItemToLoc(I, src))
+				output = I
+				to_chat(user, "<span class='notice'>You insert an output beaker.</span>")
+				return TRUE
+	return ..()
```
Comment (spookydonut) [inline code/game/objects/structures/watercloset.dm]: why change this from else and then add all the returns above?

Comment (Tlaltecuhtli) [reply code/game/objects/structures/watercloset.dm]: its easier to read

**Comment by Tlaltecuhtli** [comment]
@HomocidlePancakes centrifuge force

**Comment by HomocidlePancakes** [comment]
@Tlaltecuhtli Ok. How do you set it up so it doesn’t get diluted by toilet water, doesn’t flip or hell, where does the filtered reagent even end up?

**Comment by Tlaltecuhtli** [comment]
in the beaker you place inside after the papers

File: code/game/objects/structures/watercloset.dm (hunk 4 above)
Comment (spookydonut) [reply code/game/objects/structures/watercloset.dm]: so code churn

File: code/game/objects/structures/watercloset.dm [hunk 5]
```diff
@@ -483,4 +528,4 @@
 	icon_type = "bounty"
 	icon_state = "bounty-open"
 	color = null
-	alpha = 255
+	alpha = 255
```
Comment (Urumasi) [inline code/game/objects/structures/watercloset.dm]: EOL

**Comment by Qustinnus** [comment]
can you actually post in your PR what the fuck your PR does? You're adding a reagent made from alcohol in a toilet to make a free mood booster, so put that in the description.

**Comment by carlarctg** [comment]
peethanol is fetish content

**Comment by optimumtact** [comment]
how about you just add a ghetto centrifuge made by adding parts to a  washing machine

**Pull request closed without merging**